src/sidebar/store/modules/frames.js

```diff
@@ -4,30 +4,36 @@ import * as util from '../util';
 
 function init() {
   // The list of frames connected to the sidebar app
-  return [];
+  return {
+    frames: [],
+    // A map of frame `uri` => boolean — denotes that annotation fetching is
+    // complete for a given frame (by `uri`)
+    frameAnnotationFetchStatus: {},
+  };
 }
 
 const update = {
   CONNECT_FRAME: function (state, action) {
-    return [...state, action.frame];
+    return { frames: state.frames.concat(action.frame) };
```

Comment:
In the `DESTROY_FRAME` action, any `frameAnnotationFetchStatus` entry for a frame is explicitly removed, but `CONNECT_FRAME` does not add an entry. This is on purpose, but let me know if it feels like not the right move.